components/sup/src/manager/service/mod.rs

```diff
@@ -183,6 +193,51 @@ impl Service {
         })
     }
 
+    pub fn load(spec: ServiceSpec,
+                gossip_listen: &GossipListenAddr,
+                http_listen: &http_gateway::ListenAddr)
+                -> Result<Service> {
+        let mut ui = UI::default();
+        let package = match PackageInstall::load(&spec.ident, Some(&Path::new(&*FS_ROOT_PATH))) {
+            Ok(package) => {
+                match spec.update_strategy {
+                    UpdateStrategy::AtOnce | UpdateStrategy::Rolling => {
+                        try!(util::pkg::maybe_install_newer(&mut ui, &spec, package))
```

Comment:
That's super! I tried to preserve *existing behavior* as much as possible, so I'm proud of myself that I succeeded, heh 😄 

I agree, we didn't have this quite right. Is it easier to follow up with a commit (or PR) to make this correct so there's a commit and words around it, or roll it into this commit and rebase/push?